tests/chainermn_tests/datasets_tests/test_mnist.py

```diff
@@ -32,21 +32,22 @@ def __call__(self, x):
         return self.l3(h2)
 
 
-def check_mnist(use_gpu, use_chainerx, display_log=True):
+def check_mnist(use_gpu, use_chx, display_log=True):
     epoch = 5
     batchsize = 100
     n_units = 100
     warnings.filterwarnings(action='always', category=DeprecationWarning)
 
     model = L.Classifier(MLP(n_units, 10))
     comm = chainermn.create_communicator('naive')
+
     if use_gpu:
-        device = testing.get_device(comm.intra_rank, use_chainerx)
-        device.use()
-        model.to_device(device)
-    else:
-        device = testing.get_device(use_chainerx=use_chainerx)
+        # Call CuPy's `Device.use()` to force cudaSetDevice()
+        get_device(comm.intra_rank, False).use()
 
+    device = get_device(comm.intra_rank if use_gpu else None, use_chx)
+    device.use()
```

Comment:
Unnecessary

Reply:
I consulted ChainerX guys and found that this `device.use()` is necessary. 
ChainerX has its own device context (which means ChainerX calls `cudaSetDevice()` before and after each operation, so the original CUDA device context is preserved. Thus, we need to set both of ChainerX device context and CUDA's global device context to work with MPI.

Reply:
Hmmm, I see. That is troublesome, and confusing. We have the call `use` on twice on different context to set the same thing.

Reply:
Same here, the code is unnecessary.

Reply:
fixed.